src/test/scala/chiselTests/RawModuleSpec.scala

```diff
@@ -170,4 +170,74 @@ class RawModuleSpec extends ChiselFlatSpec with Utils with MatchesAndOmits {
       }
     }
   }
+
+  "RawModule with afterModuleBuilt" should "be able to create other modules" in {
+    val chirrtl = ChiselStage.emitCHIRRTL(new RawModule {
+      override def desiredName = "Foo"
+      val port0 = IO(Input(Bool()))
+
+      afterModuleBuilt {
+        Definition(new RawModule {
+          override def desiredName = "Bar"
+          val port1 = IO(Input(Bool()))
+        })
+      }
+    })
+
+    matchesAndOmits(chirrtl)(
+      "module Foo",
+      "input port0 : UInt<1>",
+      "module Bar",
+      "input port1 : UInt<1>"
+    )()
+  }
+
+  "RawModule with afterModuleBuilt" should "be able to instantiate the surrounding module" in {
+    @instantiable
+    class Foo extends RawModule {
+      override def desiredName = "Foo"
+      @public val port0 = IO(Input(Bool()))
+      @public val port1 = IO(Output(Bool()))
+      port1 := port0
+
+      afterModuleBuilt {
+        val fooDef = this.toDefinition
+        Definition(new RawModule {
+          override def desiredName = "Bar1"
+          val port2 = IO(Input(Bool()))
+          val foo = Instance(fooDef)
+          foo.port0 := port2
+        })
```

Comment:
Does this _need_ to be a `Definition`? What happens if this is wrapped with a `Module.apply`?

Reply:
I don't think so. This should just work as well with `Module`. Let me swap that around.